components/lab-job-scheduler/backend/src/lab_job_scheduler/job_deployer.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+from croniter import croniter
+from contaxy.operations.components import ComponentOperations
+from typing import List, Dict
+from lab_job_scheduler.schema import ScheduledJob
+import json
+from loguru import logger
+from typing import Optional
+
+
+def run_scheduled_jobs(cached_scheduled_jobs: Dict[str, Dict[str, ScheduledJob]], component_manager: ComponentOperations):
+    """Runs all scheduled jobs."""
+    for project_id, jobs in cached_scheduled_jobs.items():
+        for job_id, job in jobs.items():
+            if is_due(job):
+                logger.info(f"Deploying job {job_id}")
+                deploy_job(job, component_manager, project_id)
+                update_last_run(job, component_manager,
+                                project_id, cached_scheduled_jobs)
+                update_next_run(job, component_manager,
+                                project_id, cached_scheduled_jobs)
```

Comment:
Do this in one function to only have one DB update

Reply:
Resolved in 8aebfcb9215024e06ee83502c46e9be9a263be8e